src/core/agent/services/identifierService.ts

```diff
@@ -210,45 +205,50 @@ class IdentifierService extends AgentService {
     backgroundTask = false
   ): Promise<CreateIdentifierResult> {
     if (!identifierTypeThemes.includes(metadata.theme)) {
-      throw new Error(`${IdentifierService.THEME_WAS_NOT_VALID}`);
+      throw new Error(IdentifierService.THEME_WAS_NOT_VALID);
     }
 
     let name = `${metadata.theme}:${metadata.displayName}`;
     if (metadata.groupMetadata) {
       const initiatorFlag = metadata.groupMetadata.groupInitiator ? "1" : "0";
       name = `${metadata.theme}:${initiatorFlag}-${metadata.groupMetadata.groupId}:${metadata.displayName}`;
     }
-    let namesPendingCreation = [name];
 
     const pendingIdentifiersRecord = await this.basicStorage.findById(
       MiscRecordId.IDENTIFIERS_PENDING_CREATION
     );
 
     if (pendingIdentifiersRecord && !backgroundTask) {
-      if (!Array.isArray(pendingIdentifiersRecord.content.queuedDisplayNames)) {
+      const { queuedDisplayNames } = pendingIdentifiersRecord.content;
+
+      if (!Array.isArray(queuedDisplayNames)) {
         throw new Error(IdentifierService.INVALID_QUEUED_DISPLAY_NAMES_FORMAT);
       }
-      if (pendingIdentifiersRecord.content.queuedDisplayNames.includes(name)) {
+      if (queuedDisplayNames.includes(name)) {
         throw new Error(`${IdentifierService.IDENTIFIER_NAME_TAKEN}: ${name}`);
       }
-      namesPendingCreation = [
-        ...pendingIdentifiersRecord.content.queuedDisplayNames,
-        name,
-      ];
+
+      queuedDisplayNames.push(name);
+      await this.basicStorage.createOrUpdateBasicRecord(
+        new BasicRecord({
+          id: MiscRecordId.IDENTIFIERS_PENDING_CREATION,
+          content: { queuedDisplayNames },
+        })
+      );
+    } else {
+      await this.basicStorage.createOrUpdateBasicRecord(
+        new BasicRecord({
+          id: MiscRecordId.IDENTIFIERS_PENDING_CREATION,
+          content: { queuedDisplayNames: [name] },
+        })
+      );
     }
 
     const witnesses = await this.getAvailableWitnesses();
     if (witnesses.length === 0) {
       throw new Error(IdentifierService.NO_WITNESSES_AVAILABLE);
     }
 
-    await this.basicStorage.createOrUpdateBasicRecord(
-      new BasicRecord({
-        id: MiscRecordId.IDENTIFIERS_PENDING_CREATION,
-        content: { queuedDisplayNames: namesPendingCreation },
-      })
-    );
-
     // @TODO - foconnor: Follow up ticket to pick a sane default for threshold. Right now max is too much.
     //   Must also enforce a minimum number of available witnesses.
     const operation = await this.props.signifyClient
```

Comment:
This call will fail if you try to do it twice. You have to catch the error for 400 already incepted. The API returns 400 immediately, not in the operation. I see you're testing the operation.

I need you to temporarily throw an error at the end of this function while doing manual testing, so we can test the idempotency instead of guessing. And then remove the temporary error before commiting.

Reply:
I tried this, but we have an issue with it. We can’t directly catch errors from the API when calling `identifiers().create()`. This always returns an `op response`; if an error occurs, it will appear within this `op`.
I think we should accept handling idempotency within the operation.
![image](https://github.com/user-attachments/assets/0505a0a6-cc53-414c-b360-889b440c51e0)

Reply:
Yes, you are right it has to handle the errors within the operation op

Reply:
taking this up with Daniel in the community, it doesn't work either way, tricky error to catch and messes things up. Will see if we can await within the function